lib/galaxy/managers/collections.py

```diff
@@ -618,7 +621,9 @@ def __load_element(self, trans, element_identifier, hide_source_items, copy_elem
             decoded_id = int(trans.app.security.decode_id(encoded_id))
             hda = self.hda_manager.get_accessible(decoded_id, trans.user)
             if copy_elements:
-                element = self.hda_manager.copy(hda, history=history or trans.history, hide_copy=True, flush=False)
+                element: Union[
+                    model.HistoryDatasetAssociation, model.LibraryDatasetDatasetAssociation
+                ] = self.hda_manager.copy(hda, history=history or trans.history, hide_copy=True, flush=False)
```

Comment:
```suggestion
                element: model.HistoryDatasetAssociation = self.hda_manager.copy(hda, history=history or trans.history, hide_copy=True, flush=False)
```

Reply:
This doesn't work, on line 635 `element` is set to a `LibraryDatasetDatasetAssociation`